app/controllers/user/masquerades_controller.rb

```diff
@@ -0,0 +1,14 @@
+class User::MasqueradesController < Devise::MasqueradesController
```

Comment:
The action here is after clicking the masquerade button, `#show` finds the user to masquerade and sets current_user to that user.